protocol/communities/check_permissions_response.go

```diff
@@ -23,19 +23,21 @@ type HighestRoleResponse struct {
 	Criteria  []*PermissionTokenCriteriaResult       `json:"criteria"`
 }
 
-var joiningRoleOrders = map[protobuf.CommunityTokenPermission_Type]int{
-	protobuf.CommunityTokenPermission_BECOME_MEMBER:       1,
-	protobuf.CommunityTokenPermission_BECOME_ADMIN:        2,
-	protobuf.CommunityTokenPermission_BECOME_TOKEN_MASTER: 3,
-	protobuf.CommunityTokenPermission_BECOME_TOKEN_OWNER:  4,
+var roleOrders = map[protobuf.CommunityTokenPermission_Type]int{
+	protobuf.CommunityTokenPermission_BECOME_MEMBER:             1,
+	protobuf.CommunityTokenPermission_CAN_VIEW_CHANNEL:          2,
+	protobuf.CommunityTokenPermission_CAN_VIEW_AND_POST_CHANNEL: 3,
+	protobuf.CommunityTokenPermission_BECOME_ADMIN:              4,
+	protobuf.CommunityTokenPermission_BECOME_TOKEN_MASTER:       5,
+	protobuf.CommunityTokenPermission_BECOME_TOKEN_OWNER:        6,
```

Comment:
you changed the order. Does it require fixing the UI now? Will it impact the mobile team?